core/controllers/base.py

```diff
@@ -108,31 +108,26 @@ class NotLoggedInException(Exception):
 
         pass
 
-
```

Comment:
Are we expecting to have two newlines between two class defination?

Reply:
No, as these classes are nested, so we don't need two newlines here. This is similar to functions if they are inside a class we need a single line between them and if they are not inside a class, we need two newlines between them.

Reply:
Awesome, thanks for the explanation!